sdk/nodejs/automation/cmd.ts

```diff
@@ -37,26 +46,140 @@ export class CommandResult {
+     * @param opts.version the version of the CLI. Defaults to the CLI version matching the SDK version.
+     * @param opts.root the directory to install the CLI in. Defaults to $HOME/.pulumi/versions/$VERSION.
+     */
+    static async install(opts?: PulumiCommandOptions): Promise<PulumiCommand> {
+        const optsWithDefaults = withDefaults(opts);
+        try {
+            return await PulumiCommand.get(optsWithDefaults);
+        } catch (err) {
+            // ignore
+        }
+
+        if (process.platform === "win32") {
+            await PulumiCommand.installWindows(optsWithDefaults);
+        } else {
+            await PulumiCommand.installPosix(optsWithDefaults);
+        }
+
+        return await PulumiCommand.get(optsWithDefaults);
+    }
+
+    private static async installWindows(opts: Required<PulumiCommandOptions>): Promise<void> {
+        const response = await got("https://get.pulumi.com/install.ps1");
+        const script = await writeTempFile(response.body, { extension: ".ps1" });
+
+        try {
+            const command = process.env.SystemRoot
+                ? path.join(process.env.SystemRoot, "System32", "WindowsPowerShell", "v1.0", "powershell.exe")
+                : "powershell.exe";
+
+            const args = [
+                "-NoProfile",
+                "-InputFormat",
+                "None",
+                "-ExecutionPolicy",
+                "Bypass",
+                "-File",
+                script.path,
+                "-NoEditPath",
+                "-InstallRoot",
+                opts.root,
+                "-Version",
+                `${opts.version}`,
+            ];
+
+            await exec(command, args);
+        } finally {
+            script.cleanup();
+        }
+    }
+
+    private static async installPosix(opts: Required<PulumiCommandOptions>): Promise<void> {
+        const response = await got("https://get.pulumi.com/install.sh");
+        const script = await writeTempFile(response.body);
+
+        try {
+            const args = [script.path, "--no-edit-path", "--install-root", opts.root, "--version", `${opts.version}`];
+
+            await exec("/bin/sh", args);
+        } finally {
+            script.cleanup();
+        }
+    }
+
+    /** @internal */
+    public run(
+        args: string[],
+        cwd: string,
+        additionalEnv: { [key: string]: string },
+        onOutput?: (data: string) => void,
+    ): Promise<CommandResult> {
+        // all commands should be run in non-interactive mode.
+        // this causes commands to fail rather than prompting for input (and thus hanging indefinitely)
+        if (!args.includes("--non-interactive")) {
+            args.push("--non-interactive");
+        }
+
+        // Prepend root/bin to the path to ensure we pickup the matching
+        // bundled plugins.
+        if (this.root) {
+            additionalEnv["PATH"] = prependRootBinToPath(this.root, additionalEnv["PATH"] || process.env["PATH"]);
+        }
```

Comment:
As mentioned, this should not be needed. The CLI already correctly looks in the executable directory.

Reply:
Julian pinged me that this is correct because of multiple installs.